lib/new_relic/agent/instrumentation/thread/chain.rb

```diff
@@ -0,0 +1,24 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require_relative 'instrumentation'
+
+module NewRelic::Agent::Instrumentation
+  module Thread
+    module Chain
+      def self.instrument!
+        ::Thread.class_eval do
+          include NewRelic::Agent::Instrumentation::Thread
+
+          alias_method :initialize_without_new_relic, :initialize
+
+          def initialize(*args, &block)
+            traced_block = add_thread_tracing(*args, block)
```

Comment:
I was surprised to see `NewRelic::Agent::Instrumentation::Thread#add_thread_tracing` take `block` as a second argument instead of `&block` and then I see that it's because of the way it's being passed here. Would it be possible to use `add_thread_tracing(*args, &block)` here (prefixing `block` with `&`) and then update the method definition and block usage code for `add_thread_tracing` accordingly?

I think there's likely only a slight chance that the block isn't guaranteed to be present, and you're already guarding against `nil` by checking that `#call` is available, but I wonder if using `&block` would match other block passing patterns already in use.

Reply:
Sure, I can totally change it to be that way instead!
The only reason i didn't do that originally is because since i'm making a new block to wrap around it, I'm not treating it how we typically deal with blocks in the agent, so I just had it be a normal arg instead.